test/service/test_jobs.py

```diff
@@ -347,17 +347,19 @@ def test_throws_when_database_insertion_fails(self):
 
     def test_throws_when_algorithm_not_found(self):
         self.mock_get_algo.side_effect = bfapi.service.algorithms.NotFound('test-algo-id')
+        self.mock_get_scene.return_value = create_scene()
+        self.mock_requests.post('/tides', text=RESPONSE_TIDE)
         with self.assertRaises(jobs.PreprocessingError):
             jobs.create(API_KEY, 'test-user-id', 'test-scene-id', 'test-algo-id', 'test-name')
 
     def test_throws_when_scene_not_found(self):
-        self.mock_get_algo.return_value = create_algorithm()
+        self.mock_get_algo.return_value = create_algorithm('pzsvc-ndwi-py')
```

Comment:
`pzsvc-ndwi-py` is the default value for the `create_algorithm` function.  You don't need to repeat it here.